test/mitmproxy/test_dns.py

```diff
@@ -186,8 +186,23 @@ def assert_eq(m: dns.Message, b: bytes) -> None:
             + b"\x00\x00\x03\x84\x00H\x07ns-1707\tawsdns-21\x02co\x02uk\x00\x11awsdns-hostmaster\x06amazon\xc0\x19\x00"
             + b"\x00\x00\x01\x00\x00\x1c \x00\x00\x03\x84\x00\x12u\x00\x00\x01Q\x80\x00\x00)\x02\x00\x00\x00\x00\x00\x00\x00"
         )
-        assert invalid_rr_domain_name.answers[0].data == b"\x99live\xc0\x12"
-
+        assert (
+            invalid_rr_domain_name.answers[0].data == b"\x99live\x06github\x03com\x00"
+        )
```

Comment:
I didn't understand the purpose of this test🤔 and I'm not sure if this change is correct either

Reply:
@meitinger, thoughts? 😅

Reply:
The purpose of this test is the following: Since the original code performed a crude heuristic to check if a packed domain name is contained in the RR, it would indeed find the `0b110000000` byte (yes, pointers above an offset of 255 bytes don't trigger the old heuristic :) and try to expand the RR as domain name. But then it would encounter a `0x99` size specifier, which is not allowed (`0b10...` and `0x01...` are reserved) and abort all together. Since the new code would just ignore the first segment and still expand the `\xc0\x12` into github.com, updating the test is fine. (If we want that kind of behavior.)